### PR TITLE
Calendar for Availabilities Input

### LINKED ISSUE #20: Meeting Times as Days
## Feature
- Add the ability to click on 'Day' Button, setting the availabilities table to a calendar, so the admin can select whole days instead of times.

### PR DESCRIPTION
**Pull request by jduong09**
## Feature
- Add Calendar for user to select whole days as meeting days instead of selecting time blocks.
- Closes #20

### CODE REVIEW

**Comment by jduong09** [comment]
### Update
- Added MonthlyCalendar, which displays a calendar with no times, only the days in the month.
- User can select which days they want to meet.
- Added ability to toggle between meeting for times, and meeting for a day.
- Currently, choosing to meet for times will not allow to meet for a day, and vise-versa.
- Manipulated how duration is sent to frontend on the backend side.
- Added active states for visibility/testing purposes. Current design is not the final design.

File: root/client/src/components/monthlyTable/calendar.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { React } from 'react';
+import { Week } from './week.js';
+
+export const Calendar = ({ pollAvailabilities, year, month, firstDayOfMonth, daysInMonth, handleDayClick }) => {
+  const arrCalendar = [["", "", "", "", "", "", ""], ["", "", "", "", "", "", ""], ["", "", "", "", "", "", ""], ["", "", "", "", "", "", ""], ["", "", "", "", "", "", ""], ["", "", "", "", "", "", ""]];
```
Comment (ziggysauce) [inline root/client/src/components/monthlyTable/calendar.js]: what is this exactly?

also, i think you can programatically create this better using an array function

File: root/client/src/components/poll/pollForm.js [hunk 2]
```diff
@@ -34,18 +52,29 @@ export const PollForm = ({ handleSubmit, editData }) => {
       <label htmlFor='inputLocation'>Location (optional):
         <input id='inputLocation' name='pollLocation' type='text' onChange={e => { setPollLocation(e.target.value) }} placeholder='Conference Room A' value={pollLocation} />
       </label>
-      <label htmlFor='pollDuration'>Duration:
+      {!toggleDays
+      && <label htmlFor='pollDuration'>Duration:
```
Comment (ziggysauce) [inline root/client/src/components/poll/pollForm.js]: just my own preference, but usually i'd indent this 2 spaces

File: root/client/src/util/tablehelpers.js [hunk 3]
```diff
@@ -51,4 +51,91 @@ export const createPossibleTimeBlock = (startTime, duration) => {
   span.innerHTML = timeframe;
   span.classList.add('temp-timeblock', position, `h-${duration}`);
   return span;
+}
+
```
Comment (ziggysauce) [inline root/client/src/util/tablehelpers.js:106]: this is a good first implementation of a dictionary for finding dates. i think this is fine for this PR, but i do think there might be a better way to calculate all of this, would just need to think on it, but that can be scoped out as an optimization ticket afterwards, no need to stress about changing it now if it works.

File: root/client/src/components/monthlyTable/calendar.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    if (col + 1 < 7) {
+      col++;
+    } else {
+      row++;
+      col = 0;
+    }
+  }
+
+  const weeks = arrCalendar.map((week, idx) => {
+    return <Week pollAvailabilities={pollAvailabilities} year={year} month={month} week={week} key={idx} handleDayClick={handleDayClick} />
```
Comment (ziggysauce) [inline root/client/src/components/monthlyTable/calendar.js:47]: adding the `cursor-pointer` class to the day of the week for the calendar view i think would help

**Review by ziggysauce — CHANGES_REQUESTED** [review]

**Review by ziggysauce — COMMENTED** [review]
this doesn't have to be addressed in this PR since it's more UI, but on mobile this looks good, but i do think on tablet and desktop it would look better to make either the calendar bigger or to just show two months since there's the space for it

File: root/client/src/util/tablehelpers.js (hunk 3 above)
Comment (jduong09) [reply root/client/src/util/tablehelpers.js:106]: Sounds good, i'll mark this as resolved and saved this for future purposes.

File: root/client/src/components/monthlyTable/calendar.js (hunk 1 above)
Comment (jduong09) [reply root/client/src/components/monthlyTable/calendar.js]: Removed the array literal, it should look better.

File: root/client/src/components/monthlyTable/calendar.js (hunk 4 above)
Comment (jduong09) [reply root/client/src/components/monthlyTable/calendar.js:47]: Will be resolved in CSS PRs.

**Pull request merged**